uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticatorPreviousVersionImpl.java

```diff
@@ -0,0 +1,124 @@
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import com.microsoft.identity.client.ui.automation.logging.Logger;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+/**
+ * A model for interacting with the Microsoft Authenticator Broker App during UI Test
+ * when version number of Authenticator app under test is < "6.2206.3949"
+ */
+public class BrokerAuthenticatorPreviousVersionImpl extends BrokerMicrosoftAuthenticator {
+
+    private static final String TAG = BrokerAuthenticatorPreviousVersionImpl.class.getSimpleName();
+
+    @Override
+    public void performDeviceRegistration(@NonNull final String username,
+                                          @NonNull final String password,
+                                          final boolean isFederatedUser) {
+
+        Logger.i(TAG, "Performing Device Registration for the given account..");
+        performDeviceRegistrationHelper(
+                username,
+                password,
+                "com.azure.authenticator:id/email_input",
+                "com.azure.authenticator:id/register_button",
+                isFederatedUser
+        );
+
+        try {
+            // after device registration, make sure that we see the unregister btn to confirm successful
+            // registration
+            final UiObject unRegisterBtn = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                    "com.azure.authenticator:id/unregister_button"
+            );
+            Assert.assertTrue(
+                    "Microsoft Authenticator - Unregister Button appears.",
+                    unRegisterBtn.exists()
+            );
+
+            Assert.assertTrue(
+                    "Microsoft Authenticator - Unregister Button is clickable.",
+                    unRegisterBtn.isClickable()
+            );
+
+            // after device registration, make sure that the current registration upn matches with
+            // with what was passed in
+            final UiObject currentRegistration = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                    "com.azure.authenticator:id/current_registered_email"
+            );
+
+            Assert.assertTrue(
+                    "Microsoft Authenticator - Registered account info appears.",
+                    currentRegistration.exists()
+            );
+
+            Assert.assertTrue(
+                    "Microsoft Authenticator - Registered account upn matches provided upn.",
+                    currentRegistration.getText().equalsIgnoreCase(username)
+            );
+        } catch (final UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    public void performSharedDeviceRegistration(@NonNull final String username,
```

Comment:
Is this overriden? Override annotation?

Reply:
added this annotation in the latest commit 409b0ee9b1ff1f1b4f95466b8300946dcf71e57b